### PR TITLE
Remove Repeater.connection_settings foreign key constraint

### PR DESCRIPTION
**Pull request by millerdev**
This is preparation for moving repeaters models to a separate PostgreSQL database.

Deployed to staging.

:fish: Review by commit.

### Safety story

Tests have been added and improved to cover new functionality and ensure related object protection is preserved.

### Automated test coverage

Yes.

### QA Plan

No QA.

### Migrations

- [x] The migrations in this code can be safely applied first independently of the code

### Rollback instructions

If this PR must be rolled back the migration should be reversed as well to re-create the foreign key constraint on any environment where the migration was applied.

### CODE REVIEW

**Review by AmitPhulera — APPROVED** [review]

File: corehq/motech/repeaters/management/commands/populate_repeater_names.py [hunk 1]
```diff
@@ -16,13 +17,17 @@ def handle(self, *args, **options):
...
+            }
             for repeater in repeaters:
-                repeater.name = repeater.connection_settings.name
+                repeater.name = connections[repeater.connection_settings_id].name
                 repeater.save()
             repeaters = self._get_repeaters_with_empty_names()
 
     def _get_repeaters_with_empty_names(self):
-        return Repeater.objects.filter(name=None).select_related('connection_settings')[:CHUNKSIZE]
+        return Repeater.objects.filter(name=None)[:CHUNKSIZE]
```
Comment (kaapstorm) [inline corehq/motech/repeaters/management/commands/populate_repeater_names.py:30]: ... alternatively, we could just delete this management command. iirc it was to backfill repeater names.

**Review by kaapstorm — APPROVED** [review]

File: corehq/motech/repeaters/management/commands/populate_repeater_names.py (hunk 1 above)
Comment (millerdev) [reply corehq/motech/repeaters/management/commands/populate_repeater_names.py:30]: @SmittieC would you like to delete it (in a separate PR)? Seems like you might know best if it's no longer needed.

Comment (SmittieC) [reply corehq/motech/repeaters/management/commands/populate_repeater_names.py:30]: Sure I don't mind. It was just a temp management command

Comment (SmittieC) [reply corehq/motech/repeaters/management/commands/populate_repeater_names.py:30]: Actually, it seems like this is used by the `00012_create_default_names_for_repeaters` migration. I can just move this code to that migration file, then we should be able to remove this command.

**Pull request merged**